cpu/atmega2560/periph/gpio.c

```diff
@@ -86,51 +86,36 @@ static inline uint16_t _pin_addr(gpio_t pin)
     return (_port_addr(pin) - 0x02);
 }
 
-int gpio_init(gpio_t pin, gpio_dir_t dir, gpio_pp_t pullup)
+int gpio_init(gpio_t pin, gpio_mode_t mode)
 {
-    int res;
-
-    if (dir == GPIO_DIR_OUT) {
+    if (mode == GPIO_OUT) {
```

Comment:
Using `switch(mode){ case xyz:}` instead of `if()` would improve the readability because of the similarities with other periph drivers.